test/modern_bpf/test_suites/syscall_exit_suite/execveat_x.cpp

```diff
@@ -143,4 +143,113 @@ TEST(SyscallExit, execveatX)
 
 	evt_test->assert_num_params_pushed(23);
 }
+
+#if !defined(CAPTURE_SCHED_PROC_EXEC) && defined(__NR_clone3) && defined(__NR_wait4) && defined(__NR_execve)
+
+#include <linux/sched.h>
+
+TEST(SyscallExit, execveatX_success)
```

Comment:
Hi @Andreagit97 ,

`bpf_test` fails for this particular test case with:
````
[ RUN      ] SyscallExit.execveatX_success
[OUTPUT] SyscallExit.execveatX_success test
/root/git/falcosecurity/libs/test/modern_bpf/event_class/event_class.cpp:378: Failure
Failed
There is no event in the buffer.

[  FAILED  ] SyscallExit.execveatX_success (14 ms)
````
Need to further look into this. Do you have any tip/hint where to best start with debugging this?

Reply:
The best way to debug it is to check if the corresponding bpf program is called so setting a `bpf_printk` in the bpf code, I will try this on an `s390x` machine, BTW this is really strange because it didn't fail for `execve` :thinking: i am afraid `execveat` could exit as `execveat` and not `execve` on s390x, I hope i am wrong I will check it :/

Reply:
WOW, I've checked, when execveat returns it doesn't return as an `execve` like in the other architectures in `s390x` it correctly returns as an `execveat`, so I've to change a little bit the tests and the bpf program to address this use case... Thank you for reporting this!

Reply:
Thank you.. I just had the same idea but I was 15 hours behind ;-) Very nice catch!

Reply:
I think this is the first time that I am upset when something works correctly :laughing:  there is always a first time